### PR TITLE
LPS-129879 Adds the compatibility layer for App Builder

### PR DESCRIPTION
**Pull request by matuzalemsteles**
Depends on #176

Adds the basis for the compatibility layer for App Builder, the information that the application needs to know for now is only the `state`, `config`, and the `dispatch` method that are needed.

The implementation of DataLayoutBuilder had many implementations that it didn’t make sense to bring here, because it would generate more code and it would be more complicated, instead keep this layer simple just by exposing some methods and information and we will change the information on the AB side to the equivalent.

DataLayoutBuilder mirrored the state of two applications, on the Metal.js side which were the components of the Forms and the Data Engine, as this no longer exists, now there is only one state, we need to change it in App Builder to access only one `state` and `dispatch`.

Despite having this implementation in the feature flag, App Builder will not yet use the new App, so it is necessary to make some small changes on the App Builder side, sending some initial changes in another PR and leaving it to the ticket that will be in charge of making the switch.

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js [hunk 1]
```diff
@@ -24,4 +29,16 @@ export class DataDefinitionSchema extends Schema {
 	get availableLanguageIds() {
 		return this[SYMBOL_RAW].availableLanguageIds;
 	}
+
+	get dataDefinitionFields() {
+		return this[SYMBOL_RAW].dataDefinition.dataDefinitionFields;
+	}
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js:35]: For more information https://github.com/liferay-forms/liferay-portal/pull/176#discussion_r610042909

Maybe I'll send another commit tomorrow morning to end this.

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js (hunk 1 above)
Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js]: Done https://github.com/liferay-forms/liferay-portal/pull/179/commits/de038c40d0ded28aef55444777d46cf8701fd8fc

See the commit message for more information.

Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js]: cc @miltonmc and @alinedoleron

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
Just started reviewing :)

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
ci:forward

**Comment by interaminense** [comment]
Just started reviewing :)

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
Well, PR #183 passed the tests with these commits, closing to send over there, for some reason every time it is running the tests, it is standing in line for long hours and the time duration too...

**Comment by interaminense** [comment]
resent to: https://github.com/liferay-forms/liferay-portal/pull/183

**Pull request closed without merging**